File: DesignPatternsExercises/java/ChainOfResponsibilityDesignPattern/Solution/ChainOfResponsibilityDesignPattern.java
enum IssueType { Configuration, General, Production }

public class ChainOfResponsibilityDesignPattern {
	public static void main(String[] args) {
		SupportChain generalSupport = new GeneralSupport();
		SupportChain configurationSupport = new ConfigurationSupport();
		SupportChain productionSupport = new ProductionSupport();

		// generalSupport.__________(1)__________(configurationSupport);
		generalSupport.setNext(configurationSupport);
		// configurationSupport.__________(2)__________(productionSupport);
		configurationSupport.setNext(productionSupport);

		generalSupport.handleIssue(IssueType.Production);
	}
}

File: DesignPatternsExercises/ruby/chain_of_responsibility_design_pattern/exercise/chain_of_responsibility_design_pattern.rb
require_relative 'configuration_support'
require_relative 'general_support'
require_relative 'production_support'

general_support = GeneralSupport.new
configuration_support = ConfigurationSupport.new
production_support = ProductionSupport.new

general_support.next(configuration_support)
configuration_support.next(production_support)

general_support.handle_issue(:production)
